### PR TITLE
MNT Update unit test to work with both HTML4Value and HTML5Value

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripeltd/product-issues/issues/665

Fixes https://github.com/silverstripe/recipe-kitchen-sink/actions/runs/3987666536/jobs/6837802266#step:12:76
`1) SilverStripe\CMS\Tests\Model\SiteTreeTest::testSanitiseExtraMeta with data set #5 ('<link rel="canonical" href="v...ert(1)', '<link rel="canonical" href="v...gy="">', 'Invalid HTML is converted to ...parsed')
SilverStripe\ORM\ValidationException: Custom Meta Tags does not contain valid HTML`

recipe-sink will use HTML5Value (from silverstripe/html5) and installer will use the default HTML4Value

I'm honestly not sure why this only started breaking now

Have tested locally separately using both installer and recipe-kitchen-sink

### CODE REVIEW

File: tests/php/Model/SiteTreeTest.php [hunk 1]
```diff
@@ -2060,12 +2057,27 @@ public function provideSanitiseExtraMeta(): array
...
-                '<link rel="canonical" href="valid" ;;// somethingdodgy < onmouseover=alert(1)',
+            ]
+        ];
+    }
+
+    public function testSanatiseInvalidExtraMeta()
+    {
+        $siteTree = new SiteTree();
+        $siteTree->ExtraMeta = '<link rel="canonical" href="valid" ;;// somethingdodgy < onmouseover=alert(1)';
+        if (class_exists(HTML5Value::class)) {
```
Comment (maxime-rainville) [inline tests/php/Model/SiteTreeTest.php]: `HTML5Value` could exists without being the default.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
This seems like two distinct scenarios to me that should be covered by two distinct tests.

Is it practical to update the config on the spot to swap which parser is used, so that both test are run if the HTML5 module is present? If not, the test should be skip if the appropriate parser is not in used.

File: tests/php/Model/SiteTreeTest.php (hunk 1 above)
Comment (emteknetnz) [reply tests/php/Model/SiteTreeTest.php:2068]: It makes itself the default - https://github.com/silverstripe/silverstripe-html5/blob/2/_config/html.yml#L6

The only scenario where it's not the default is if someone installs silverstripe/html5 and then manually configures to use HTML4Value ... which is not ever going to be an issue in CI

Comment (emteknetnz) [reply tests/php/Model/SiteTreeTest.php]: Have updated

**Review by maxime-rainville — APPROVED** [review]
LGTM

**Pull request merged**